pkg/config/crd/resource_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crd
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+
+	cfg "istio.io/mixer/pkg/config/proto"
+)
+
+func deepEqualNonNil(v1 interface{}, v2 interface{}) bool {
```

Comment:
done. It's used in `TestConvert` below. Conversion path of map[string]interface{} -> proto.Message -> map[string]interface{} could drop some keys for nil value, this function special handles the pattern like that.